packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1342,6 +1344,35 @@ abstract class FlutterCommand extends Command<void> {
     );
   }
 
+  // This adds the Dart defines used to access various Flutter version information at runtime.
+  void _addFlutterVersion() {
+    const List<String> dartDefines = [
+      'FLUTTER_VERSION',
+      'FLUTTER_CHANNEL',
+      'FLUTTER_GIT_URL',
```

Comment:
I'm wondering if this could cause data leaking. This should be in a really low traffic but still possible?

Reply:
Yeah, I've also wondered that already. I'm not too invested in this data. I just included it to keep the `FlutterVersion` class on par with the output of `flutter --version`